test/unit/lexer/location_offset.d

```diff
@@ -0,0 +1,482 @@
+    TOK.override_ : Test("override"),
+    TOK.delegate_ : Test("delegate"),
+    TOK.function_ : Test("function"),
+    TOK.mixin_ : Test("mixin"),
+    TOK.align_ : Test("align"),
+    TOK.extern_ : Test("extern"),
+    TOK.private_ : Test("private"),
+    TOK.protected_ : Test("protected"),
+    TOK.public_ : Test("public"),
+    TOK.export_ : Test("export"),
+    TOK.static_ : Test("static"),
+    TOK.final_ : Test("final"),
+    TOK.const_ : Test("const"),
+    TOK.abstract_ : Test("abstract"),
+    TOK.debug_ : Test("debug"),
+    TOK.deprecated_ : Test("deprecated"),
+    TOK.in_ : Test("in"),
+    TOK.out_ : Test("out"),
+    TOK.inout_ : Test("inout"),
+    TOK.lazy_ : Test("lazy"),
+    TOK.auto_ : Test("auto"),
+    TOK.package_ : Test("package"),
+    TOK.immutable_ : Test("immutable"),
+
+    TOK.if_ : Test("if"),
+    TOK.else_ : Test("else"),
+    TOK.while_ : Test("while"),
+    TOK.for_ : Test("for"),
+    TOK.do_ : Test("do"),
+    TOK.switch_ : Test("switch"),
+    TOK.case_ : Test("case"),
+    TOK.default_ : Test("default"),
+    TOK.break_ : Test("break"),
+    TOK.continue_ : Test("continue"),
+    TOK.with_ : Test("with"),
+    TOK.synchronized_ : Test("synchronized"),
+    TOK.return_ : Test("return"),
+    TOK.goto_ : Test("goto"),
+    TOK.try_ : Test("try"),
+    TOK.catch_ : Test("catch"),
+    TOK.finally_ : Test("finally"),
+    TOK.asm_ : Test("asm"),
+    TOK.foreach_ : Test("foreach"),
+    TOK.foreach_reverse_ : Test("foreach_reverse"),
+    TOK.scope_ : Test("scope"),
+    TOK.onScopeExit : Test("scope exit", "scope(exit)"),
+    TOK.onScopeFailure : Test("scope failure", "scope(failure)"),
+    TOK.onScopeSuccess : Test("scope success", "scope(success)"),
+
+    TOK.invariant_ : Test("invariant"),
+
+    TOK.unittest_ : Test("unittest"),
+
+    TOK.argumentTypes : Test("__argTypes"),
+    TOK.ref_ : Test("ref"),
+    TOK.macro_ : Test("macro"),
+
+    TOK.parameters : Test("__parameters"),
+    TOK.traits : Test("__traits"),
+    TOK.overloadSet : Test("__overloadset"),
+    TOK.pure_ : Test("pure"),
+    TOK.nothrow_ : Test("nothrow"),
+    TOK.gshared : Test("__gshared"),
+
+    TOK.line : Test("__LINE__"),
+    TOK.file : Test("__FILE__"),
+    TOK.fileFullPath : Test("__FILE_FULL_PATH__"),
+    TOK.moduleString : Test("__MODULE__"),
+    TOK.functionString : Test("__FUNCTION__"),
+    TOK.prettyFunction : Test("__PRETTY_FUNCTION__"),
+
+    TOK.shared_ : Test("shared"),
+    TOK.at : Test("at sign", "@"),
+    TOK.pow : Test("power", "^^"),
+    TOK.powAssign : Test("power assign", "^^="),
+    TOK.goesTo : Test("fat arrow", "=>"),
+    TOK.vector : Test("__vector"),
+    TOK.pound : Test("pound", "#"),
+
+    TOK.interval : Test("interval"),
+    TOK.voidExpression : Test("voidexp"),
+    TOK.cantExpression : Test("cantexp"),
+    TOK.showCtfeContext : Test("showCtfeContext"),
+
+    TOK.objcClassReference : Test("class"),
+    TOK.vectorArray : Test("vectorArray"),
+];
+
+static foreach (test; tests)
```

Comment:
The point of using an associative array was to iterate the members of the `TOK` enum. Then use each value as a key into the associative array. This will make sure that if a new member is added to `TOK`, this test will fail.

Reply:
Something like this:
`static foreach (tok; TOK)`
raises an `invalid foreach aggregate` error
Is there a way to make this work?

Reply:
Yes, it's possible. I'm trying to come up with some good pointers without directly telling you the how the code should look like. I think it will be more beneficial I you manage to find the answer yourself. Should be possible to search for this online.

Reply:
I did find something about EnumMembers but would it be alright to rely on phobos or is there some equivalent in dmd?

Reply:
Should be fine. Alternatively, you can use `__traits(allMembers)` as well [1].

[1] https://dlang.org/spec/traits.html#allMembers.